include/SSVOpenHexagon/Components/CWall.hpp

```diff
@@ -67,12 +68,40 @@ class CWall
         return curve;
     }
 
+    [[gnu::always_inline, nodiscard]] bool isOverlapping(
+        const sf::Vector2f& mPoint, const sf::Vector2f& mCenterPos,
+        const float mRadiusSquared) noexcept
+    {
+        // If the wall is too far from the center is cannot cause collision.
+        // If it is and both distance and pointInPolygon() calculations are executed
+        // we lose performance, If not (which is the case in the vast majority of
+        // any level's runtime) we only do a much faster algorithm.
+        if(ssvs::getMagSquared((vertexPositions[0] + vertexPositions[1]) / 2.f -
+            mCenterPos) > mRadiusSquared)
+        {
+            outOfPlayerRadius = true;
+            return false;
+        }
```

Comment:
This seems like a nice little optimization. Have you actually measured the impact? Would be interesting to know. 
Also wondering if we can avoid the call to `getMagSquared` somehow and avoid the extra multiplication, maybe we could cache that value once per frame?

Reply:
I am not sure how we could avoid using getMagSquared(), I am already skipping the unnecessary step of running sqrt on it since if x > y => x * x > y * y but that is as far as I think we can go without running in inaccurate calculations.
Caching would provide no benefits since that specific overload is called only the first collision check of a frame, the second one relies on
```
    [[gnu::always_inline, nodiscard]] bool isOverlapping(
        const sf::Vector2f& mPoint) const noexcept
    {
        if(outOfPlayerRadius)
        {
            return false;
        }

        return Utils::pointInPolygon(vertexPositions, mPoint.x, mPoint.y);
    }
```
I will see to measure the impact one of the next days.

Reply:
Example code returns 2 0
` ` ` 
#include <iostream>
#include <chrono>
#include <array>

struct Point
{
    float x;
    float y;
};

inline bool pointInPolygon(const std::array<Point, 4>& points, float x, float y) noexcept
{
    bool result{false};

    for(unsigned int i{0}, j{3}; i < 4; j = i++)
    {
        const auto& vI{points[i]};
        const auto& vJ{points[j]};

        if(((vI.y > y) != (vJ.y > y)) &&
            (x < (vJ.x - vI.x) * (y - vI.y) / (vJ.y - vI.y) + vI.x))
        {
            result = !result;
        }
    }

    return result;
}

inline float magSquared(Point& p) noexcept
{
    return p.x * p.x + p.y * p.y;
}

int main()
{
    auto t1 = std::chrono::high_resolution_clock::now();
    std::array<Point, 4> points = {Point{0.f, 0.f}, Point{5.f, 0.f}, Point{10.f, 22.f}, Point{-2.f, 6.f}};
    const auto b = pointInPolygon(points, 4.f, 6.f);
    auto t2 = std::chrono::high_resolution_clock::now();
    std::cout << std::chrono::duration_cast<std::chrono::microseconds>( t2 - t1 ).count() << std::endl;

    auto t3 = std::chrono::high_resolution_clock::now();
    Point p = {20.f, 16.f};
    const auto f = magSquared(p);
    auto t4 = std::chrono::high_resolution_clock::now();
    std::cout << std::chrono::duration_cast<std::chrono::microseconds>( t4 - t3 ).count() << std::endl;

    return 0;
}
` ` `